### ISSUE TITLE
catching callback errors

### ISSUE THREAD

**Issue by charlierudolph** [atom/season#8]
season is catching an error thrown by one of my callbacks which made debugging a pain.
The offending function is:
```coffee
parseContents = (objectPath, cachePath, contents, callback) ->
  try
    object = parseObject(objectPath, contents)
    writeCacheFile(cachePath, object) if cachePath
    callback?(null, object)
  catch parseError
    callback?(parseError)
```
try/catches should be extremely limited in their scope to they only catch errors they want to.
Thus I believe this function needs to be updated to:
```coffee
parseContents = (objectPath, cachePath, contents, callback) ->
  try
    object = parseObject(objectPath, contents)
  catch parseError
    return callback?(parseError)

  writeCacheFile(cachePath, object) if cachePath
  callback?(null, object)
```
I'm happy to create a PR with this change and test verifying this behavior if you agree

**Comment by kevinsawicki** [comment]
> I'm happy to create a PR with this change and test verifying this behavior if you agree

That would be great, thanks so much :+1:

**Comment by charlierudolph** [comment]
So the only way I can think to test this is by stubbing out the readFile to make it synchronous. However I do not know how to do that with jasmine (nor am I interested). I work with mocha, chai, and sinon. Thus sorry I have to leave this for someone else to implement

**Comment by kevinsawicki** [comment]
Thanks for reporting this, fixed and published in 5.1.5
